tests/unit/test_policies.py

```diff
@@ -460,27 +498,37 @@ def test_no_live_nodes(self):
             h.set_location_info("dc1", "rack1")
             hosts.append(h)
 
-        policy = DCAwareRoundRobinPolicy("dc1", used_hosts_per_remote_dc=1)
+        policy = policy_specialization(*constructor_args, used_hosts_per_remote_dc=1)
         policy.populate(Mock(), hosts)
 
         for host in hosts:
             policy.on_down(host)
 
         qplan = list(policy.make_query_plan())
-        self.assertEqual(qplan, [])
+        assert qplan == []
 
-    def test_no_nodes(self):
+    def test_no_nodes(self, policy_specialization, constructor_args):
         """
         Ensure query plan for an empty cluster will execute without errors
         """
 
-        policy = DCAwareRoundRobinPolicy("dc1", used_hosts_per_remote_dc=1)
+        policy = policy_specialization(*constructor_args, used_hosts_per_remote_dc=1)
         policy.populate(None, [])
 
         qplan = list(policy.make_query_plan())
-        self.assertEqual(qplan, [])
+        assert qplan == []
+
+    def test_wrong_dc(self, policy_specialization, constructor_args):
+        hosts = [Host(DefaultEndPoint(i), SimpleConvictionPolicy) for i in range(3)]
+        for h in hosts[:3]:
+            h.set_location_info("dc2", "rack2")
 
-    def test_default_dc(self):
+        policy = policy_specialization(*constructor_args, used_hosts_per_remote_dc=0)
+        policy.populate(Mock(), hosts)
+        qplan = list(policy.make_query_plan())
+        assert len(qplan) == 0
+
+    def test_default_dc(self, policy_specialization, constructor_args):
         host_local = Host(DefaultEndPoint(1), SimpleConvictionPolicy, 'local')
         host_remote = Host(DefaultEndPoint(2), SimpleConvictionPolicy, 'remote')
         host_none = Host(DefaultEndPoint(1), SimpleConvictionPolicy)
```

Comment:
This test only uses `DCAwareRoundRobinPolicy`, ignores specialization, so it should probably be outside the class.